### PR TITLE
Delimited integration test

### PR DESCRIPTION
**Pull request by bluemonk3y**
#434 - to provide coverage and validation. Includes extending IntegrationTestHarness to support other Serdes

### CODE REVIEW

File: ksql-engine/src/test/java/io/confluent/ksql/integration/DelimUdfIntTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+@Category({IntegrationTest.class})
+public class DelimUdfIntTest {
+
+  private IntegrationTestHarness testHarness;
+  private KsqlContext ksqlContext;
+  private Map<String, RecordMetadata> recordMetadataMap;
+  private String topicName = "TestTopic";
+  private TestDataProvider dataProvider;
+
+  String format = DataSource.DataSourceSerDe.DELIMITED.name();
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/integration/DelimUdfIntTest.java]: nit: inline?

File: ksql-engine/src/test/java/io/confluent/ksql/integration/DelimUdfIntTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            "ID, DESCRIPTION",
+            "ITEMS",
+            "ID LIKE '%_1'"
+    );
+
+    ksqlContext.sql(queryString);
+
+    Map<String, GenericRow> expectedResults = Collections.singletonMap("ITEM_1", new GenericRow(Arrays.asList( "ITEM_1", "home cinema")));
+
+    Map<String, GenericRow> results = testHarness.consumeData(testStreamName, dataProvider.schema(), 1, new StringDeserializer(), IntegrationTestHarness.RESULTS_POLL_MAX_TIME_MS);
```
Comment (dguy) [inline ksql-engine/src/test/java/io/confluent/ksql/integration/DelimUdfIntTest.java]: Should we add retries here similar to what has been done in `WindowingIntTest`, i.e, 
```
TestConditions.waitForCondition(() -> {
     // return true if test condition satisfied
}, timeout, "failure message")
```

Comment (bluemonk3y) [reply ksql-engine/src/test/java/io/confluent/ksql/integration/DelimUdfIntTest.java]: Im not sure its needed - we aren't windowing, just pulling from a stream

Comment (bluemonk3y) [reply ksql-engine/src/test/java/io/confluent/ksql/integration/DelimUdfIntTest.java]: if it proves flakey I will add it in...

**Pull request merged**